### PR TITLE
test(anchor-navigation): typescript refactor

### PR DESCRIPTION
**Pull request by stephenogorman**
### Proposed behaviour

- Rename the `anchor-navigation.cy.js` to `anchor-navigation.cy.tsx` file in `./cypress/components/anchor-navigation/` folder.
- Refactor tests to Typescript.  

### Current behaviour

Anchor-navigation tests are in js not ts.

### Checklist

- [x] Commits follow our style guide
- [ ] Related issues linked in commit messages if required
- [ ] Screenshots are included in the PR if useful
- [ ] All themes are supported if required
- [ ] Unit tests added or updated if required
- [x] Cypress automation tests added or updated if required
- [ ] Storybook added or updated if required
- [ ] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [ ] Typescript `d.ts` file added or updated if required
- [ ] Related docs have been updated if required

#### QA

- [x] Tested in CodeSandbox/storybook
- [x] Add new Cypress test coverage if required
- [x] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### Testing instructions
- [ ] Run `npx cypress open --component` to check if the`anchor-navigation.cy.tsx` file passed
- [ ] Run `npx cypress run --component` to check none of the other *.cy.tsx files have regressed

### CODE REVIEW

File: cypress/components/anchor-navigation/anchor-navigation.cy.tsx [hunk 1]
```diff
@@ -55,7 +55,7 @@ context("Testing AnchorNavigation component", () => {
     it("should pass accessibility tests for Anchor Navigation in full screen dialog", () => {
       CypressMountWithProviders(<InFullScreenDialogStory />);
 
-      getDataElementByValue("main-text")
+      return getDataElementByValue("main-text")
```
Comment (DlgSHi) [inline cypress/components/anchor-navigation/anchor-navigation.cy.tsx:58]: why do we need to return this?

Comment (stephenogorman) [reply cypress/components/anchor-navigation/anchor-navigation.cy.tsx:58]: I was getting a number of Typescript errors yesterday before I did an update on VS, and this solved one of them. I'm not getting them any more so I'll remove this.

Comment (stephenogorman) [reply cypress/components/anchor-navigation/anchor-navigation.cy.tsx:58]: I was seeing the following error on that piece of code:
"Promise should be returned to test its fulfillment or rejection"
Adding the return fixed it.

**Review by DlgSHi — APPROVED** [review]

**Pull request merged**